### ISSUE TITLE
[Rule Error.] Modifiers rules for stepping dice up and down.

### ISSUE THREAD

**Issue by slfrnkln** [Stefouch/t2k4e#45]
### Describe the Bug

When applying modifiers to the dice rolls they still sometimes work off the alpha logic rather than the new release logic.
ie. They will increase the bigger die first before increasing the smaller die.

This however was modified in the release rules. When modifying dice you should aim to _balance_ the dice rather than increase the gap between them.

[Edit] It seems to be when there are modifiers of >1 or when the normal roll only includes one dice.

### Severity

- High (a broken feature, major functional impact)

### Context

Any skill roll with modifiers

### To Reproduce

Click a skill and apply a modifier.

Example:
Actor with d8 EMP and d8 Command:❌
1. Click Command
2. Apply a +2 modifier
3. Roll

Actor with d8 EMP and d8 Command:❌
1. Click Command
2. Apply a -2 modifier
3. Roll

Actor with d8 EMP and d6 Command:✔
1. Click Command
2. Apply a +1 modifier
3. Roll

Actor with d8 EMP and d6 Command:✔
1. Click Command
2. Apply a -1 modifier
3. Roll

Actor with d8 EMP and no Command:❌
1. Click Command
2. Apply a +2 modifier
3. Roll

### Expected Behavior

Actor with d8 EMP and d8 Command:
When the +2 modifier is applied it should roll d10, d10 (instead rolls d8, d12)❌

Actor with d8 EMP and d8 Command:
When the -2 modifier is applied it should roll d6, d6 (instead rolls d8)❌

Actor with d8 EMP and d6 Command:
When the +1 modifier is applied it should roll d8, d8 (is correct, rolls d8, d8)✔

Actor with d8 EMP and d6 Command:
When the -1 modifier is applied it should roll d6, d6 (is correct, rolls d6, d6)✔

Actor with d8 EMP and no Command:
When the +1 modifier is applied it should roll d8, d6 (instead rolls d10) ❌

### Screenshots

![image](https://user-images.githubusercontent.com/53422981/132659399-673332b9-f2e5-4cd7-89fe-232ac8ef2236.png)

**Comment by lystraeudaimon** [comment]
![SKILL CHECKS](https://user-images.githubusercontent.com/90668658/133192753-2436a1b1-3d77-42e4-a458-7ead4ee6eda4.png)
I've been using this chart for my in person sessions. I think it's accurate.

**Comment by Stefouch** [comment]
There is indeed a big misbehavior of the dice modifier algorithm.
I have asked Free League for more details and to confirm the chart.
Thank you all for the thorough report. I'll work on a fix asap.

**Comment by Kayne** [comment]
I made a pull request which fixes the calculation of dice (#63).
Basically it increases or decreases die one by one so the balance is preserved.

**Comment by Kayne** [comment]
I just updated FoundryVTT to 9 and updated module to latest version (2.0.2) and there is one case where dice are not properly calculated: when you have just one dice and modifies it. 

For example, one D6 with modifier +2 should result in D8 and D6, but instead, rolls d10 which is wrong (looks like it just adds two dice "upgrades" to D6 which is incorrect). It should add new dice to make 2D6 and then upgrade one of the D6 to D8.

Also, no dice with modifier plus 2 results with D10 but it should be 2D6. However this roll should not happen anyhow, but I was just interested in the outcome.

I tested all other cases and everything else looks fine.

**Comment by Kayne** [comment]
Fixing current behavior was easy. I just created pull request #79 with minor code change which correctly adds die to the pool when modifier is positive and only one die in the pool.

I did not fix issue with empty pool with positive modifier since such roll should not be requested by player anyhow, so I'm not sure if is it worth to invest time on fixing it. However, if you are interested in this… How to reproduce? No dice in pool, modifier set to +1 and roll. The effect is script rolls two dice (instead of one D6), because dice pool is always populated with 1 die and then modifier is applied. As I wrote earlier, it's not a big issue because I do not think that empty pool dice is even possible by the rules. I think player always has at least one die.
